File: Server/src/main/java/com/example/agecalculator/AgeCalculatorClient.java
package com.example.agecalculator;

import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;
import com.example.agecalculator.AgeCalculatorGrpc.AgeCalculatorBlockingStub;

public class AgeCalculatorClient {

    public static void main(String[] args) {
        // Create a channel to connect to the server
        ManagedChannel channel = ManagedChannelBuilder.forAddress("localhost", 50051)
                .usePlaintext()  // Disable TLS to allow insecure communication
                .build();

        // Create a stub (blocking - synchronous)
        AgeCalculatorBlockingStub stub = AgeCalculatorGrpc.newBlockingStub(channel);

        // Create a request with the birth year
        int birthYear = 1990; // You can modify this as needed
        AgeCalculatorProto.AgeRequest request = AgeCalculatorProto.AgeRequest.newBuilder()
                .setBirthYear(birthYear)
                .build();

        // Call the service and get the response
        AgeCalculatorProto.AgeReply response = stub.tellAge(request);

        // Print the response
        System.out.println(response.getMessage());

        // Shutdown the channel
        channel.shutdown();
    }
}

File: Client/age_calculator_client.py
import grpc
import age_calculator_pb2
import age_calculator_pb2_grpc

def run():
    # Open a gRPC channel
    with grpc.insecure_channel('localhost:50051') as channel:
        stub = age_calculator_pb2_grpc.AgeCalculatorStub(channel)

        # Create request with the birth year
        birth_year = input('Enter birth year: ')
        request = age_calculator_pb2.AgeRequest(birthYear=int(birth_year))

        # Make the call
        response = stub.TellAge(request)
        print(response.message)


if __name__ == '__main__':
    run()
